doc_template/examples_root/examples/biophysical_sim/biophysical_sim.py

```diff
@@ -0,0 +1,125 @@
+from allensdk.model.biophysical import runner,_schemas
+import json
+import os
+from argschema import ArgSchemaParser
+from allensdk.api.queries.biophysical_api import BiophysicalApi
+from allensdk.core.nwb_data_set import NwbDataSet
+import numpy as np
+import matplotlib.pyplot as plt
+import seaborn as sns
+import shutil
+
+sns.set(style='whitegrid',font_scale=1.5)
+plt.rcParams.update({'axes.grid':False})
+
+#%% Define utility functions
+
+def get_manifest_args(args):
+    return runner.load_description(args)
+    
+
+def get_sweep_data(nwb_file,sweep_number):
+    nwb = NwbDataSet(nwb_file)
+    sweep = nwb.get_sweep(sweep_number)
+    stim_diff = np.diff(sweep['stimulus']*1e12)
+    stim_start = np.where(stim_diff != 0)[0][-2]
+    stim_end = np.where(stim_diff != 0)[0][-1]
+    
+    # read v and t as numpy arrays
+    v = sweep['response']*1e3
+    dt = 1.0e3 / sweep['sampling_rate']
+    num_samples = len(v)
+    t = np.arange(num_samples) * dt
+    return t,v,t[stim_start],t[stim_end]
+
+def copytree(src, dst, symlinks=False, ignore=None):
+    for item in os.listdir(src):
+        s = os.path.join(src, item)
+        d = os.path.join(dst, item)
+        if os.path.isdir(s):
+            shutil.copytree(s, d, symlinks, ignore)
+        else:
+            shutil.copy2(s, d)
+            
+#%% Pick a cell and a sweep to run from the available set of protocols
+            
+cell_id = 468193142 # get this from the web site: http://celltypes.brain-map.org
+sweep_num = 46 # Select a Long Square sweep :  1s DC
+
+#%% Download all-acive model
+
+sdk_model_templates = {491455321:'all_active',329230710:'perisomatic'}
+bp = BiophysicalApi()
+bp.cache_stimulus = True 
+model_list = bp.get_neuronal_models(cell_id)
+model_dict = {}
+for model_dict_ in model_list:
+    if model_dict_['neuronal_model_template_id'] in sdk_model_templates.keys():
+        model_dict[sdk_model_templates[model_dict_['neuronal_model_template_id']]] = model_dict_['id']
+
+model_dir = 'all_active_models'
+bp.cache_data(model_dict['all_active'], working_directory=model_dir)
+new_model_file = 'fit_parameters_new.json'
+shutil.copyfile(new_model_file,os.path.join(model_dir,new_model_file))
+copytree('modfiles',os.path.join(model_dir,'modfiles'))
+
+#%% Running the legacy all-active models
+
+os.chdir(model_dir)
```

Comment:
could this be avoided by building up a path?

Reply:
you mean changing the basedir in the config file? The api always downloads the model files in a relative directory so unless we change the config file we need to cd into the downloaded dir.